src/mono/mono/mini/aot-compiler.c

```diff
@@ -12948,13 +12948,21 @@ should_emit_extra_method_for_generics (MonoMethod *method, gboolean reference_ty
 	MonoGenericParam *gen_param = NULL;
 	unsigned int gen_param_count;
 
+	ERROR_DECL (error);
```

Comment:
Add an extra `MonoError*` argument to `should_emit_extra_method_for_generics` and to `should_emit_gsharedvt_method` and pass in an error from the caller (it's `collect_methods` for both).

If the error is not `is_ok (error)` at the call sites, you need to call `mono_error_cleanup` on it to deallocate it.  (if the error is `ok`, there's nothing to cleanup, but it also is harmless to call `mono_error_cleanup`)